xarray/convert.py

```diff
@@ -49,3 +58,81 @@ def set_cdms2_attrs(var, attrs):
     cdms2_var = cdms2.createVariable(var.values, axes=axes, id=dataarray.name)
     set_cdms2_attrs(cdms2_var, var.attrs)
     return cdms2_var
+
+
+# TODO: Add converting bounds from xarray to Iris and back
+# TODO: Cell methods are not converted between Iris and xarray
+def to_iris(dataarray):
+    """Convert a DataArray into a Iris Cube
+    """
+    # Iris not a hard dependency
+    import iris
+    # iris.unit is deprecated in Iris v1.9
+    import cf_units
+
+    def check_attrs(attrs, keys):
+        return dict((k, v) for k, v in attrs.items() if k in keys)
+
+    def get_args(attrs):
+        _args = {'attributes': filter_attrs(attrs, iris_forbidden_keys)}
+        _args.update(check_attrs(attrs, ('standard_name', 'long_name',)))
+        _unit_args = check_attrs(coord.attrs, ('calendar',))
+        if attrs.has_key('units'):
+            _args['units'] = cf_units.Unit(attrs['units'], **_unit_args)
+        return _args
+
+    dim_coords = []
+    aux_coords = []
+
+    for coord_name in dataarray.coords:
+        coord = encode(dataarray.coords[coord_name])
+        coord_args = get_args(coord.attrs)
+        coord_args['var_name'] = coord_name
+        iris_coord = iris.coords.DimCoord(coord.values, **coord_args)
```

Comment:
Yes I had that thought too. I started off with the code only working for dimensional coordinates and then extended to auxiliary later. It's because `add_aux_coord` can accept `DimCoord` or `AuxCoord` and my test is only 1D. I think it'll probably fail if the coord is 2D with the way it's currently written. I will check.